litestar/utils/sync.py

```diff
@@ -81,7 +81,7 @@ def set_parsed_signature(self, namespace: dict[str, Any]) -> None:
         """
         from litestar.utils.signature import ParsedSignature
 
-        self._parsed_signature = ParsedSignature.from_fn(self.ref.value, namespace)
+        self._parsed_signature = ParsedSignature.from_fn(unwrap_partial(self.ref.value), namespace)
```

Comment:
I found we were generating the parsed signature on the `AsyncPartial` instance, causing the return of the `before_request` handlers to always appear as `Any`.

Whether this PR is merged, or not, this should go into main.